multipart/multipart.py

```diff
@@ -117,6 +125,35 @@ def parse_options_header(value):
     return ctype, options
 
 
+class Cache(Generic[T]):
+    __slots__ = ('_value', '_is_set')
+
+    def __init__(self):
+        self._value: Optional[T] = None
+        self._is_set: bool = False
+
+    @property
+    def value(self) -> Optional[T]:
+        if self.is_set:
+            return self._value
+        else:
+            raise ValueError("Value not yet set")
+
+    @value.setter
+    def value(self, v: T):
+        self._value = v
+        self._is_set = True
+
+    def clear(self):
+        """Reset the cache"""
+        self._value = None
+        self._is_set = False
+
+    def is_set(self) -> bool:
+        """Check if value has been set"""
+        return self._is_set
```

Comment:
Can we wait for this kind of change till we have a better test coverage?

Reply:
Your call. I believe I added this to fix type-hinting for `Field.__repr__` only, so it is not important.

Reply:
I'd prefer for this PR to not have runtime changes, unless really needed.